### PR TITLE
91691

### PR DESCRIPTION
**Pull request by jayreddy519**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [91691](https://devops.dchbx.org/redmine/issues/91691) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | n/a |
| Peer Review Story Points | n/a |
| Ticket Author | @karashearer5 |

### CODE REVIEW

File: app/controllers/insured/plan_shoppings_controller.rb [hunk 1]
```diff
@@ -192,6 +192,7 @@ def generate_eligibility_data
     if shopping_tax_household.present? && @hbx_enrollment.coverage_kind == 'health' && @hbx_enrollment.kind == 'individual'
       @tax_household = shopping_tax_household
       @max_aptc = @tax_household.total_aptc_available_amount_for_enrollment(@hbx_enrollment, @hbx_enrollment.effective_on)
+      @hbx_enrollment.update_attributes(aggregate_aptc_amount: @max_aptc)
```
Comment (saikumar9) [inline app/controllers/insured/plan_shoppings_controller.rb:195]: Please update the **aggregate_aptc_amount** on enrollment at the same place where we were updating **applied_aptc_amount** on the enrollment.

File: app/domain/operations/individual/apply_aggregate_to_enrollment.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        generate_enrollments    = yield generate_enrollments(eligible_enrollments, params[:eligibility_determination])
+        Success(generate_enrollments)
+      end
+
+      private
+
+      def validate(params)
+        return Failure("Given object is not a valid eligibility determination object") unless params[:eligibility_determination].is_a?(EligibilityDetermination)
+        return Failure("No active tax household for given eligibility") unless params[:eligibility_determination].tax_household.present?
+        Success(params[:eligibility_determination])
```
Comment (saikumar9) [inline app/domain/operations/individual/apply_aggregate_to_enrollment.rb:24]: Add validation to return a failure if the given enrollment non-ivl or dental enrollment.

File: spec/domain/operations/individual/apply_aggregate_to_enrollment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Operations::Individual::ApplyAggregateToEnrollment do
```
Comment (saikumar9) [inline spec/domain/operations/individual/apply_aggregate_to_enrollment_spec.rb:5]: Good test coverage.

**Review by saikumar9 — CHANGES_REQUESTED** [review]
Please review the comments.

File: app/controllers/insured/plan_shoppings_controller.rb (hunk 1 above)
Comment (ymhari) [reply app/controllers/insured/plan_shoppings_controller.rb:195]: Sai, as we discuss we should be updating the aggregate amount for shopping enrollments/during shopping as well and shopping enrollment does not necessarily have to go till the confirmation page where we update the applied aptc amount on enrollment.

File: app/domain/operations/individual/apply_aggregate_to_enrollment.rb (hunk 2 above)
Comment (ymhari) [reply app/domain/operations/individual/apply_aggregate_to_enrollment.rb:24]: Sai, we are not sending enrollment as param here. Instead we fetch them in the operation where we take of these requirements through the scopes.

**Review by saikumar9 — APPROVED** [review]

**Pull request merged**